### PR TITLE
Make forbidden apis check cacheable and cc compatible

### PR DESCRIPTION
**Pull request by breskeby**
Also ported the implementation to use the gradle worker api that allows us to expand our parallalism of task execution within a single project. Example build scan: https://gradle-enterprise.elastic.co/s/non7hf532jrlg/timeline?sort=longest&type=org.elasticsearch.gradle.internal.precommit.CheckForbiddenApisTask

### CODE REVIEW

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/DependencyLicensesTask.java [hunk 1]
```diff
@@ -88,8 +88,6 @@ public class DependencyLicensesTask extends DefaultTask {
 
     private final Logger logger = Logging.getLogger(getClass());
 
-    private static final String SHA_EXTENSION = ".sha1";
```
Comment (breskeby) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/DependencyLicensesTask.java:91]: not used; removed while been there

**Review by mark-vieira — APPROVED** [review]

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/ForbiddenApisPrecommitPlugin.java [hunk 2]
```diff
@@ -65,60 +52,36 @@ public TaskProvider<? extends Task> createTask(Project project) {
             String sourceSetTaskName = sourceSet.getTaskName(FORBIDDEN_APIS_TASK_NAME, null);
             var sourceSetTask = project.getTasks().register(sourceSetTaskName, CheckForbiddenApisTask.class, t -> {
                 t.setDescription("Runs forbidden-apis checks on '${sourceSet.name}' classes.");
+                t.setResourcesDir(resourcesDir);
                 t.getOutputs().upToDateWhen(Specs.SATISFIES_ALL);
                 t.setClassesDirs(sourceSet.getOutput().getClassesDirs());
                 t.dependsOn(resourcesTask);
                 t.setClasspath(sourceSet.getRuntimeClasspath().plus(sourceSet.getCompileClasspath()).plus(sourceSet.getOutput()));
```
[the remark this answers is not in the document]
Comment (breskeby) [reply build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/ForbiddenApisPrecommitPlugin.java]: I recall I explicitly added the getOutput() to this quite a while ago. I don't recall why that is needed. The compileclasspath is required as we have a lot of compileOnly dependencies declared and we need those to run forbiddenapis check successfully.

[the remark this answers is not in the document]
Comment (mark-vieira) [reply build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/ForbiddenApisPrecommitPlugin.java]: Fair enough.

**Comment by breskeby** [comment]
This work was related to https://github.com/elastic/elasticsearch/issues/57918

**Pull request merged**